.devcontainer/Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM ghcr.io/zephyrproject-rtos/ci-repo-cache:v0.26.13.20240601
```

Comment:
ci-repo-cache is intended for internal use by the upstream CI and [contains a special repository cache under `/repo-cache`](https://github.com/zephyrproject-rtos/docker-ci-repo-cache/blob/main/Dockerfile).

For local development use, it is recommended to use the [`zephyr-build` image](https://github.com/zephyrproject-rtos/docker-image/pkgs/container/zephyr-build).

```suggestion
FROM ghcr.io/zephyrproject-rtos/zephyr-build:v0.26.13
```

Reply:
Thanks for the link. I had already read those Dockerfiles though, I specially picked the image with the cache in order to speed up the checkout of the workspace. The tradeoff is mentioned in the commit message.

`zephyr-build` is a superset of the `ci` image, which means it's even bigger and has stuff like a whole x11, window manager (openbox) and vnc. I'm not sure we want that in a cloud-based container workspace.

Creating a codespace fails not because of the repo cache, but when the layers with the toolchains are being added. The base `ci` image is already too big (I saw something like ~3.9GB) apparently for github's builder.  I just now tried to use `zephyr-build` [here](https://github.com/jori-nordic/zephyr/commit/327bb8a0c749de3b51e22a58e7423eef958ba579) and have the same out of space issue (see image at bottom).

I understand why `ci` and `zephyr-build` have all the toolchains and that it would be messy to split into multiple images/layers. Like what toolchains are "essential", what are extra etc. We struggled quite a bit at Nordic on this, where and how to layer the images,etc.
I also don't know if the cache we have on the zephyr homegrown cloud would work well with a more layered approach. 

That said, and even if it is hard, I think we should try splitting the main image into multiple sub-images. Just like it is possible to not have to check out all the HALs from vendors we don't need in a given project.
IMO that will be very good for UX, not just for enabling devs to package up their dev environment, but will also make spinning up a per-project, versioned CI easier. 

I'm talking from the PoV of application writers, not developers of zephyr. If you already thought about this and we have reasons to not split the images, that's fine. It would just be nice if we could.

![image](https://github.com/user-attachments/assets/c030e92b-b46c-4696-b5a9-b41052e72ca6)

Reply:
maybe as an idea have a base image layer that can build arch_posix and native_sim, and layer other toolchains by big families (e.g. ARM, intel, riscv) on top?

For my (admittedly special) use-case, I only need babblesim + a native compiler. Plus the deps of check_compliance and the zephyr build system.

It'd be nice if others could chime in on what they need.

Reply:
I guess what @stephanosio is saying here, do not use the repo cache, use a different repo with the same containers. the cache is for internal use only in CI. The size issue needs to be resolved somewhere else.

Reply:
I see, if I really want the cache for development use then I should make my own, is that it?

I don't think it saves that much time (over a normal `west update`) anyways, as the lifetime of the devcontainer will be much larger than ephemeral CI containers. I'll change the image right away :+1: 

What do you think about merging the current PR and fixing the size issue later? It is still usable locally, we used it just now with theo to debug a failing test without having to bork my local tree. 
I can add a caveat to the "codespaces" section in the docs.